common/src/main/java/me/shedaniel/clothconfig2/api/AbstractConfigEntry.java

```diff
@@ -54,6 +55,12 @@ public abstract class AbstractConfigEntry<T> extends DynamicElementListWidget.El
     private List<String> cachedTags = null;
     private Iterable<String> additionalSearchTags = null;
     
+    @Nullable private Requirement enableRequirement = null;
+    @Nullable private Requirement displayRequirement = null;
+    
+    private final AtomicBoolean enabled = new AtomicBoolean(true);
```

Comment:
It's being set on the main thread and read on the render thread, but you're right that shouldn't lead to any race conditions...

Thinking about it, there could be some weird edge-cases if the value is set mid-thread; such that some render methods treat the entry as enabled while others treat it as disabled...

But Atomic doesn't solve that. Would need to lock the value for the duration of our rendering code... Which is probably overkill for single-frame visual edge-cases.

Reply:
We could avoid the enabled state changing mid-frame by using a frame-value cache, along the lines of:

```java
public class FrameValue<T> {
    
    // The latest value
    private T value;
    
    // The value currently in use, updated by calling sync()
    private T cached;
    
    public FrameValue(@NotNull T value) {
        this.value = this.cached = value;
    }
    
    public void set(@NotNull T value) {
        this.value = value;
    }
    
    public void sync() {
        this.cached = value;
    }
    
    public @NotNull T get() {
        return this.cached;
    }
}
```

`set()` would be called from the main thread and `get()` called multiple times per frame on the render thread.

`sync()` would be called once per frame, preferably at the start or end of the rendering. Not sure the best way to actually do that, or if its worth it to avoid single-frame race-conditions that are minor and aesthetic in nature...

Reply:
Switched to primitive boleans for now.

Reply:
Yeah a simple primitive boolean should do, I think tick() is actually ran on the Render Thread, both render and tick are done in Minecraft.runTick().